### PR TITLE
Browser compile

### PR DESCRIPTION
**Pull request by adamdbradley**
`stencil.compile(code)`

### CODE REVIEW

File: src/compiler/app-core/format-component-runtime-meta.ts [hunk 1]
```diff
@@ -2,15 +2,15 @@ import * as d from '../../declarations';
 import { CMP_FLAGS, LISTENER_FLAGS, MEMBER_FLAGS } from '@utils';
 
 
-export function formatLazyBundleRuntimeMeta(bundleId: any, cmps: d.ComponentCompilerMeta[]): d.LazyBundleRuntimeData {
+export const formatLazyBundleRuntimeMeta = (bundleId: any, cmps: d.ComponentCompilerMeta[]): d.LazyBundleRuntimeData => {
```
Comment (manucorporat) [inline src/compiler/app-core/format-component-runtime-meta.ts]: this code does not run in the browser, right? why transform to arrow now?

File: src/compiler/browser/browser-scope-css.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import * as d from '../../declarations';
+import { catchError } from '@utils';
+import { getScopeId } from '../style/scope-css';
+import { scopeCss as shadowCss } from '../../utils/shadow-css';
+
+
+export const scopeCss = async (code: string, tagName: string, opts: d.ScopeCssOptions = {}) => {
```
Comment (manucorporat) [inline src/compiler/browser/browser-scope-css.ts]: a lot of functions with the same name, can get tricky to maintain in the future

File: src/compiler/transformers/core-runtime-apis.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  Host: `Host as ${HOST}`,
+  HTMLElement: HTML_ELEMENT,
+  proxyCustomElement: `proxyCustomElement as ${PROXY_CUSTOM_ELEMENT}`,
+  registerHost: `registerHost as ${REGISTER_HOST}`,
+  registerInstance: `registerInstance as ${REGISTER_INSTANCE}`,
+};
+
+
+export const addCoreRuntimeApi = (moduleFile: d.Module, coreRuntimeApi: string) => {
+  if (!moduleFile.coreRuntimeApis.includes(coreRuntimeApi)) {
```
Comment (manucorporat) [inline src/compiler/transformers/core-runtime-apis.ts]: using Set?

Comment (manucorporat) [inline src/compiler/transformers/core-runtime-apis.ts]: using Set?

File: src/runtime/initialize-component.ts [hunk 4]
```diff
@@ -78,15 +78,19 @@ export const initializeComponent = async (elm: d.HostElement, hostRef: d.HostRef
...
+
+      const shadowCssPolyfill = (BUILD.shadowDom && cmpMeta.$flags$ & CMP_FLAGS.needsShadowDomShim);
+      const runtimeScopedCss = (BUILD.runtimeScopeCss && cmpMeta.$flags$ & CMP_FLAGS.scopedCssEncapsulation);
+
+      if (!BUILD.hydrateServerSide && (shadowCssPolyfill || runtimeScopedCss)) {
         style = await import('../utils/shadow-css').then(m => m.scopeCss(style, scopeId, false));
       }
       registerStyle(scopeId, style, !!(cmpMeta.$flags$ & CMP_FLAGS.shadowDomEncapsulation));
-      Cstr.isStyleRegistered = true;
+      Cstr.$isStyleRegistered$ = true;
```
Comment (manucorporat) [inline src/runtime/initialize-component.ts]: love it

File: src/runtime/vdom/vdom-render.ts [hunk 5]
```diff
@@ -532,8 +532,8 @@ const relocateSlotContent = (
 
               // add to our list of nodes to relocate
               relocateNodes.push({
-                slotRefNode: childNode,
-                nodeToRelocate: node
+                $slotRefNode$: childNode,
```
Comment (manucorporat) [inline src/runtime/vdom/vdom-render.ts]: good caught!

File: src/compiler/app-core/format-component-runtime-meta.ts (hunk 1 above)
Comment (adamdbradley) [reply src/compiler/app-core/format-component-runtime-meta.ts]: Not everything is going to run in the browser, but in updating the ones that are I probably updated some others too. And I guess for consistency

File: src/compiler/browser/browser-scope-css.ts (hunk 2 above)
Comment (adamdbradley) [reply src/compiler/browser/browser-scope-css.ts]: Yeah the shadow-css.ts file and fn needs to be renamed.

File: src/compiler/transformers/core-runtime-apis.ts (hunk 3 above)
Comment (adamdbradley) [reply src/compiler/transformers/core-runtime-apis.ts]: moduleFile needs to be serializable so we can cache it.

**Pull request closed without merging**